backend/src/main/kotlin/org/loculus/backend/service/submission/SubmissionDatabaseService.kt

```diff
@@ -332,7 +337,7 @@ class SubmissionDatabaseService(
                 },
             ) {
                 it[processingStatusColumn] = PROCESSED.name
-                it[processedDataColumn] = compressionService.compressSequencesInProcessedData(processedData, organism)
+                it[processedDataColumn] = compressionService.compressSequencesInProcessedData(dehydratedProcessedData, organism)
```

Comment:
Might actually make sense to compress first, dehydrate second - or even dehydrate as part of the compression service

Reply:
Don't have strong feelings here, happy with whatever others prefer - in some sense removing nulls is a type of compression, albeit not on sequences, so the method name should change

Reply:
turns out that actually solves failing tests, so going down the compression route was the right choice!

Reply:
it's much neater code wise as well